dask/utils.py

```diff
@@ -2091,6 +2091,11 @@ def get_default_shuffle_algorithm() -> str:
         from distributed import default_client
 
         default_client()
-        return "tasks"
+        try:
+            import pyarrow  # noqa
+
+            return "p2p"
+        except ImportError:
+            return "tasks"
```

Comment:
We could add a warning/info log suggesting to install pyarrow if we hit this exception.

Reply:
Should we add a `p2p` extras target that requires `distributed` and a suitable `pyarrow` version?

Reply:
A p2p extra feels way too specific. I'd rather add this to the `dataframe` target but that'd obviously be a bigger change